CHANGES.md

```diff
@@ -1,6 +1,166 @@
 CHANGES
 =======
 
+4.0
+===
+
+This release contains several breaking changes. For a complete migration guide, see:
+https://django-csp.readthedocs.io/en/latest/migration-guide.html
+
+## Breaking Changes
+
+- **Configuration Format**: Moved to dict-based configuration which allows for setting policies for
+both enforced and report-only. Instead of using individual settings with `CSP_` prefixes, you now use
+dictionaries called `CONTENT_SECURITY_POLICY` and/or `CONTENT_SECURITY_POLICY_REPORT_ONLY`.
+
+  You can use Django's check command to automatically identify existing CSP settings and generate a
+  template for the new configuration format:
+
+  ```
+  python manage.py check
+  ```
+
+  This will detect your old `CSP_` prefixed settings and output a draft of the new dict-based
+  configuration, giving you a starting point for migration.
+
+  **Example:**
+
+  Change from:
+  ```python
+  CSP_DEFAULT_SRC = ["'self'", "*.example.com"]
+  CSP_SCRIPT_SRC = ["'self'", "js.cdn.com/example/"]
+  CSP_IMG_SRC = ["'self'", "data:", "example.com"]
+  CSP_EXCLUDE_URL_PREFIXES = ["/admin"]
+  ```
+
+  to:
+  ```python
+  from csp.constants import SELF
+
+  CONTENT_SECURITY_POLICY = {
+      "EXCLUDE_URL_PREFIXES": ["/admin"],
+      "DIRECTIVES": {
+          "default-src": [SELF, "*.example.com"],
+          "script-src": [SELF, "js.cdn.com/example/"],
+          "img-src": [SELF, "data:", "example.com"],
+      },
+  }
+  ```
+
+- **Nonce Configuration**: Switched from specifying directives that should contain nonces as a
+separate list to using a sentinel `NONCE` value in the directive itself.
+
+  **Example:**
+
+  Change from:
+  ```python
+  CSP_INCLUDE_NONCE_IN = ['script-src', 'style-src']
+  ```
+
+  to:
+  ```python
+  from csp.constants import NONCE, SELF
+
+  CONTENT_SECURITY_POLICY = {
+      "DIRECTIVES": {
+          "script-src": [SELF, NONCE],
+          "style-src": [SELF, NONCE],
+      }
+  }
+  ```
+
+- **Nonce Behavior**: Changed how `request.csp_nonce` works - it is now Falsy
+(`bool(request.csp_nonce)`) until it is read as a string (e.g., used in a template or with
+`str(request.csp_nonce)`). Previously, it always tested as `True`, and testing generated the nonce.
```

Comment:
Suggestion: can we explain why we did this (or link to the relevant Issue at least)?